### ISSUE TITLE
`swift package update ` command stops updating !

### ISSUE THREAD

**Issue by Jinxiansen** [vapor/vapor#1413]
🚀 Thanks for contributing! Choose from one of the templates below. [REMOVE]

When I use the **swift package update** to update the package, I always stuck in Updating https://github.com/vapor/service.git this line for up to several hours.

I guess that's because I added
  **.package (url: "https://github.com/vapor/mysql.git", .revision ("beta"))**
When I remove  **mysql.git**, it will not stop there.

* Vapor Toolbox: 3.1.2
* OS:  mac OS 10.13.2

![12](https://user-images.githubusercontent.com/16829428/34677965-bdd7a45c-f4cc-11e7-9c9c-9deff5c79679.png)

>  English is not my native language; please excuse typing errors.

**Comment by 0xTim** [comment]
@Jinxiansen this usually means that there is a problem with dependencies? Are you using Vapor 3? In which case your dependency should read `.package (url: "https://github.com/vapor/mysql.git", .branch("beta"))` since revision should point to a single commit

**Comment by vzsg** [comment]
The markdown-provider can cause similar trouble if you aren't using it's Vapor 3 specific branch.

**Comment by 0xTim** [comment]
(markdown-provider also doesn't work at the moment due to a bug in Leaf) - @Jinxiansen if you are still having issues can you post your `Package.swift`

**Comment by Jinxiansen** [comment]
@0xTim @vzsg 

My vapor is 3.1.2,

<img width="1027" alt="222" src="https://user-images.githubusercontent.com/16829428/34680029-c70eaf2e-f4d2-11e7-8765-959ea4f8e339.png">

This is my package.swift configuration, I do not understand the difference between upToNextMajor, branch and revision, the official document data rarely seldom, but also see the specific how to add and use a package .
_Please two big brother urged me! Thank you!_

> English is not my native language; please excuse typing errors.

**Comment by 0xTim** [comment]
Just to clarify your Vapor Toolbox is 3.1.2, your framework version is 2, as set by the `.package (url: "https://github.com/vapor/vapor.git", .upToNextMajor(from: "2.2.0"))`

For details on how the `Package.swift` manifest file works have a look at the [official SPM documentation](https://github.com/apple/swift-package-manager/blob/master/Documentation/PackageDescriptionV4.md)

If you are trying to include MySQL into your Vapor project and use it with Fluent you don't actually need to the MySQL repo, just the Fluent MySQL repo. Remove both the MySQL dependency and the Fluent MySQL dependency and add in `.package(url: "https://github.com/vapor/mysql-provider.git", .upToNextMajor(from: "2.0.0")),` instead.

Note that you also need to add the dependencies to your targets as discussed in the documentation above and in [my blog post](https://geeks.brokenhands.io/blog/posts/migrating-your-server-side-app-to-swift-4/). See https://github.com/brokenhandsio/SteamPressExample/blob/master/Package.swift#L23 for an example

**Comment by Jinxiansen** [comment]
@0xTim 

Oh! I saw your brother after the project, finally found the reason!
Because I only added .package in Pakcage.swift _(url: "https://github.com/vapor/mysql-provider.git", .upToNextMajor (from: "2.0.0"))_
   But did not add **dependencies** in the corresponding `**MySQLProvider**`
When I joined, and after executing `swift package update` and `swift package generate-xcodeproj`, I finally saw MySQLProvider 2.0 and MySQL 2.0.3 in the vapor project!
